src/code/z_debug.c

```diff
@@ -3,223 +3,285 @@
 
 typedef struct {
-    u16 push;
-    u16 held;
-} InputCombo;
+    u16 hold;
+    u16 press;
+} InputCombo; // size = 0x4
 
 GameInfo* gGameInfo;
-s32 D_8015FA94; // no known symbols
-PrintTextBuffer D_8015FA98[0x16];
 
-s16 D_8011E0B0 = 0; // PrintTextBuffer index
-Color_RGBA8 printTextColors[] = {
-    { 255, 255, 32, 192 }, { 255, 150, 128, 192 }, { 128, 96, 0, 64 },     { 192, 128, 16, 128 },
-    { 255, 192, 32, 128 }, { 230, 230, 220, 64 },  { 128, 150, 255, 128 }, { 128, 255, 32, 128 },
+PrintTextBufferEntry sDebugPrintTextBuffer[22];
+s16 sDebugPrintTextBufferNumUsed = 0;
+Color_RGBA8 sDebugPrintTextColors[] = {
+    { 255, 255, 32, 192 },  // 0
+    { 255, 150, 128, 192 }, // 1
+    { 128, 96, 0, 64 },     // 2
+    { 192, 128, 16, 128 },  // 3
+    { 255, 192, 32, 128 },  // 4
+    { 230, 230, 220, 64 },  // 5
+    { 128, 150, 255, 128 }, // 6
+    { 128, 255, 32, 128 },  // 7
 };
 
-InputCombo inputCombos[REG_GROUPS] = {
-    { BTN_L, BTN_CUP },    { BTN_L, BTN_CLEFT }, { BTN_L, BTN_CDOWN }, { BTN_L, BTN_A },          { BTN_R, BTN_CDOWN },
-    { BTN_L, BTN_CRIGHT }, { BTN_L, BTN_R },     { BTN_L, BTN_DLEFT }, { BTN_L, BTN_DRIGHT },     { BTN_L, BTN_DUP },
-    { BTN_L, BTN_B },      { BTN_L, BTN_Z },     { BTN_L, BTN_DDOWN }, { BTN_R, BTN_A },          { BTN_R, BTN_B },
-    { BTN_R, BTN_Z },      { BTN_R, BTN_L },     { BTN_R, BTN_CUP },   { BTN_R, BTN_CRIGHT },     { BTN_R, BTN_DLEFT },
-    { BTN_R, BTN_CLEFT },  { BTN_R, BTN_START }, { BTN_L, BTN_START }, { BTN_R, BTN_DRIGHT },     { BTN_R, BTN_DUP },
-    { BTN_START, BTN_R },  { BTN_START, BTN_A }, { BTN_START, BTN_B }, { BTN_START, BTN_CRIGHT },
+InputCombo sRegGroupInputCombos[REG_GROUPS] = {
+    { BTN_L, BTN_CUP },        //  REG
+    { BTN_L, BTN_CLEFT },      // SREG
+    { BTN_L, BTN_CDOWN },      // OREG
+    { BTN_L, BTN_A },          // PREG
+    { BTN_R, BTN_CDOWN },      // QREG
+    { BTN_L, BTN_CRIGHT },     // MREG
+    { BTN_L, BTN_R },          // YREG
+    { BTN_L, BTN_DLEFT },      // DREG
+    { BTN_L, BTN_DRIGHT },     // UREG
+    { BTN_L, BTN_DUP },        // IREG
+    { BTN_L, BTN_B },          // ZREG
+    { BTN_L, BTN_Z },          // CREG
+    { BTN_L, BTN_DDOWN },      // NREG
+    { BTN_R, BTN_A },          // KREG
+    { BTN_R, BTN_B },          // XREG
+    { BTN_R, BTN_Z },          // cREG
+    { BTN_R, BTN_L },          // sREG
+    { BTN_R, BTN_CUP },        // iREG
+    { BTN_R, BTN_CRIGHT },     // WREG
+    { BTN_R, BTN_DLEFT },      // AREG
+    { BTN_R, BTN_CLEFT },      // VREG
+    { BTN_R, BTN_START },      // HREG
+    { BTN_L, BTN_START },      // GREG
+    { BTN_R, BTN_DRIGHT },     // mREG
+    { BTN_R, BTN_DUP },        // nREG
+    { BTN_START, BTN_R },      // BREG
+    { BTN_START, BTN_A },      // dREG
+    { BTN_START, BTN_B },      // kREG
+    { BTN_START, BTN_CRIGHT }, // bREG
+
 };
 
-char regChar[] = " SOPQMYDUIZCNKXcsiWAVHGmnBdkb";
+char sRegGroupChars[REG_GROUPS] = {
+    ' ', //  REG
```

Comment:
The full string kinda makes sense to me tbh, but I guess it's debatable.
Although I would at least omit the REG name comments since the names literally come from these characters, so it's a bit redundant imo

Reply:
Yeah my idea is that it confirms the letters associated with each group, and it helps realize that this is where they come from
Also, formatting and symmetry with `sRegGroupInputCombos`

?